.buildkite/scripts/pre-install-command.sh

```diff
@@ -12,16 +33,34 @@ add_bin_path(){
     export PATH="${WORKSPACE}/bin:${PATH}"
 }
 
-with_mage() {
+with_go() {
     create_bin_folder
-    retry 5 curl -sL -o "${WORKSPACE}/bin/mage.tar.gz" "https://github.com/magefile/mage/releases/download/v${SETUP_MAGE_VERSION}/mage_${SETUP_MAGE_VERSION}_Linux-64bit.tar.gz"
```

Comment:
It looks like there is no artifact for Darwin and ARM in 1.15.0, so I changed the approach to install `mage` using `go install` command.